src/etc/inc/services.inc

```diff
@@ -369,24 +371,26 @@ function services_radvd_configure($blacklist = array()) {
 		$radvdconf .= "\t};\n";
 
 		/* add DNS servers */
-		$dnslist = array();
-		if (isset($config['dnsmasq']['enable']) || isset($config['unbound']['enable'])) {
-			$dnslist[] = $ifcfgipv6;
-		} elseif (is_array($config['system']['dnsserver']) && !empty($config['system']['dnsserver'])) {
-			foreach ($config['system']['dnsserver'] as $server) {
-				if (is_ipaddrv6($server)) {
-					$dnslist[] = $server;
+		if (($dhcpv6ifconf['enable-radvd-dns']) != 'disabled') {
```

Comment:
Same as above for `enable-dhcp6c-dns`, use `dhcp6c-dns` or something similar to avoid redundancy.

Reply:
@jim-p Names fixed